app/models/va_caution_flag.rb

```diff
@@ -16,6 +16,8 @@ class VaCautionFlag < ApplicationRecord
 
   validates :facility_code, presence: true
   validate :validate_date_fields
+  validate :validate_settlement_data
```

Comment:
Move all of this into validator class, see https://github.com/department-of-veterans-affairs/gibct-data-service/blob/master/app/models/edu_program.rb for how to setup